### PR TITLE
Affinitylib

### PR DESCRIPTION
**Pull request by jonathan-beard**
@f-tischler , Can you double check to make sure I'm not stomping on any of the updates you planned? @mr-j0nes, wouldn't mind an extra set of eyes just in case.... 

# Primary changes for this pull request are:
* move core affinity code out to separate module
* move demangle code to separate module
* update cmake files to support namespace injection
* bugfix in mapbase.hpp which could leave uninitialized portinfo objects
* move "portNotFound" code from header and place in cpp file so that demangle gets
into an object vs. being re-included in every header/raft file (would be a pain
with the namespace injection later on)
* Fix pkg-config install mechanism to make building raftlib apps far easier
* update versions of shm, affinity, demangle libs (to also include shm bug fix)

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -90,3 +100,18 @@ foreach( HFILE ${MAINHEADERS} )
  install( FILES ${CMAKE_CURRENT_SOURCE_DIR}/${HFILE}  DESTINATION ${CMAKE_INSTALL_PREFIX}/include )
 endforeach( HFILE ${MAINHEADERS} )
 
+##
+# setup pkg-config files
+##
+if( NOT ${USEQTHREAD} )
+install( CODE "file( COPY  ${CMAKE_SOURCE_DIR}/raftlib.pc
```
Comment (jonathan-beard) [inline CMakeLists.txt:107]: Change to CURRENT_SOURCE_DIR otherwise will pull from wrong DIR when submodule

Comment (jonathan-beard) [reply CMakeLists.txt]: fixed.

**Comment by f-tischler** [comment]
Checking on Windows right now

**Comment by f-tischler** [comment]
Does not build for me, include "affinity.hpp" not found in simpleschedule.cpp. Seems like the include directory is missing.

**Comment by jonathan-beard** [comment]
will revert in a moment, or see if I can find a quick fix. I need to find more reviewers with windows experience, any suggestions?  

>

**Comment by f-tischler** [comment]
@jonathan-beard there is a problem with how affinity lib is build. First it is not added to the Windows list of dependencies in gitmodules.cmake. This results in the error above. However, when adding affinity as dependency, it builds with the warning that no thread pinning is supported but then raftlib does not build because affinity lib has public dependency on pthreads which does not exist on Windows.

**Comment by f-tischler** [comment]
Conditionally linking pthreads on UNIX does fix one part of this problem

**Comment by f-tischler** [comment]
The second problem is with the affinity test suite being compiled on Windows which does not make any sense. Also, just a side note, the affinity test should not link explicitly pthreads as this happens automatically when linking to affinity.

**Comment by f-tischler** [comment]
> I need to find more reviewers with windows experience, any suggestions?

No, I am afraid not

**Comment by jonathan-beard** [comment]
Will see if the free version of visual studio is useful....if not, might just purchase it. I had hoped @2noodle  had windows :). Will put together a patch for the patch to fix the includes and builds. Will add you as reviewer.

**Comment by mr-j0nes** [comment]
A Windows workflow [action](https://github.com/features/actions) could be setup in order to know if there are errors.

https://github.com/microsoft/setup-msbuild

But I have 0 experience with that...

**Comment by jonathan-beard** [comment]
Yeah, I think @f-tischler is going to check into that. There's a patch we've gotta test to make sure there's nothing I've missed.

**Comment by f-tischler** [comment]
I will test #97 manually and setup a Github action once this PR is merged into master

**Pull request merged**